packages/flutter/lib/src/rendering/binding.dart

```diff
@@ -278,12 +278,14 @@ mixin RendererBinding on BindingBase, ServicesBinding, SchedulerBinding, Gesture
 
   @override // from GestureBinding
   void dispatchEvent(PointerEvent event, HitTestResult? hitTestResult) {
-    if (hitTestResult != null ||
-        event is PointerAddedEvent ||
-        event is PointerRemovedEvent) {
```

Comment:
Why did you think that this clause is unnecessary ?
For example `hitTestResult == null && event is PointerMoveEvent`, it seems that change some behaviours , right?

Reply:
According to https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/gestures/binding.dart#L390, 
```dart
/// The `hitTestResult` argument may only be null for [PointerAddedEvent]s or  /// [PointerRemovedEvent]s.
```

Reply:
Thanks for explanation, LGTM again!